pytest_xmlcomp/plugin.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+__version__ = "0.2.0"
+
+import pytest
+
+from py.path import local
+import json
+from lxml import etree
+import pluggy
+from pytest_xmlcomp import hooks
+
+#hookspec = pluggy.HookspecMarker("pytest_xmlcomp")
+#hookimpl = pluggy.HookimplMarker("pytest_xmlcomp")
+
+
+#pm = pluggy.PluginManager("pytest_xmlcomp")
+#pm.add_hookspecs(hooks.TransformSpec)
+#pm.register(hooks.TransformXML())
+
+def pytest_addoption(parser):
+    group = parser.getgroup('xmlcomp')
+    group.addoption(
+            '--datadir',
+            action='store',
+            dest='datadir',
+            default='tests/data',
+            help='Give the directory containing the XML files.')
+
+def pytest_addhooks(pluginmanager):
+    pluginmanager.add_hookspecs(hooks)
+    pluginmanager.register(hooks, "pytest")
+
+def pytest_collect_file(parent, path):
+    jsonfile = path.new(ext = ".json")
+    if path.ext == ".xml" and jsonfile.exists():
+        print("XML file %s and JSON file %s found" % (path, jsonfile))
+        return XMLJSONFile(path, parent)
+
+
+def stringifylist(res):
+        result = []
+        for obj in res:
+            if isinstance(obj, (etree._Comment, etree._ProcessingInstruction)):
+                result.append(str(obj))
+            elif isinstance(obj, etree._ElementTree):
+                result.append('/') # FIXME
+            elif isinstance(obj, etree._Element):
+                result.append("<%s>" %obj.tag)
+            elif isinstance(obj, str):
+                result.append(obj)
+        return result
+
+
+class XMLJSONFile(pytest.File):
+    def collect(self):
+        from lxml import etree
+        import json
+        originalxmlfile = self.fspath
+        pm = self.config.pluginmanager
+        modifiedxmlfile = pm.hook.pytest_transform_XML(xmlfile = originalxmlfile)
+        try:
+            tree = etree.parse(str(modifiedxmlfile[0]))
+        except etree.XMLSyntaxError:
+            print("XML Syntax Error in file %s" % self.fspath)
```

Comment:
As it is a syntax error, we should use one of the the function `pytest.warn()` or `pytest.fail()`. The last could be a bit too destructive, as we maybe don't want to stop the testing. 

Hmn, maybe we should use an option for that?